### PR TITLE
Refactor query param

### PR DESCRIPTION
**Pull request by noCharger**
### Description
1. Refactor SparkSubmitParameters
2. Move query param from `spark/src/main/java/org/opensearch/sql/spark/client/StartJobRequest.java` into `spark/src/main/java/org/opensearch/sql/spark/asyncquery/model/SparkSubmitParameters.java` to avoid input limitation from EMRS

### Issues Resolved
https://github.com/opensearch-project/sql/issues/2376
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] ~New functionality has been documented.~
  - [ ] ~New functionality has javadoc added~
  - [ ] ~New functionality has user manual doc added~
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by noCharger** [comment]
Confirmed the BWC failure unrelated to this PR by reproducing the same issue with a shadow PR checked from main branch https://github.com/opensearch-project/sql/actions/runs/7894433419/job/21545080478?pr=2520

File: spark/src/main/java/org/opensearch/sql/spark/asyncquery/model/SparkSubmitParameters.java [hunk 1]
```diff
@@ -86,35 +64,101 @@ public Builder clusterName(String clusterName) {
...
+    }
+
+    public Builder extraParameters(String params) {
+      this.extraParameters = params;
+      return this;
     }
 
-    private void setFlintIndexStoreHost(URI uri) {
+    public Builder query(String query) {
+      config.put(FLINT_JOB_QUERY, query);
```
Comment (noCharger) [inline spark/src/main/java/org/opensearch/sql/spark/asyncquery/model/SparkSubmitParameters.java:83]: TODO: 1. wrap with "" 2. escape special chars like (

**Comment by noCharger** [comment]
Rebased to main, @vamsi-amazon  @penghuo  @dai-chen  Please review

**Review by penghuo — APPROVED** [review]

**Review by dai-chen — APPROVED** [review]

**Review by penghuo — APPROVED** [review]

**Comment by seankao-az** [comment]
documenting some findings on an issue with batch flint job on EMR-S after this PR.

Batch job submitted to EMR-S will fail.

For clearer error message, modified this line:
https://github.com/opensearch-project/opensearch-spark/blob/e6a97dcc8c248788b7afc7843f90282f4f8db7c4/spark-sql-application/src/main/scala/org/apache/spark/sql/FlintJob.scala#L50
into
```
throw new IllegalArgumentException(
  s"""Unsupported number of arguments. Expected 1 or 2 arguments. ${args.length}, ${args.mkString(" ")}""")
```

From the error message, perhaps it has something to do with how EMR parse the Spark Submit Parameters

##### The entire stderr file for the batch job
Example 1:
```
--conf spark.flint.job.query=refresh skipping index on mys3.default.http_logs
```
```
Files  s3://flint-dev-seankao/opensearch-spark-jars/opensearch-spark-standalone_2.12-0.3.0-SNAPSHOT.jar from /tmp/spark-b28da86c-b55e-45e6-9aa5-48432a15c19e/opensearch-spark-standalone_2.12-0.3.0-SNAPSHOT.jar to /home/hadoop/./opensearch-spark-standalone_2.12-0.3.0-SNAPSHOT.jar
Files  s3://flint-dev-seankao/opensearch-spark-jars/opensearch-spark-sql-application_2.12-0.3.0-SNAPSHOT.jar from /tmp/spark-b28da86c-b55e-45e6-9aa5-48432a15c19e/opensearch-spark-sql-application_2.12-0.3.0-SNAPSHOT.jar to /home/hadoop/./opensearch-spark-sql-application_2.12-0.3.0-SNAPSHOT.jar
Files  s3://flint-dev-seankao/opensearch-spark-jars/opensearch-spark-ppl_2.12-0.3.0-SNAPSHOT.jar from /tmp/spark-b28da86c-b55e-45e6-9aa5-48432a15c19e/opensearch-spark-ppl_2.12-0.3.0-SNAPSHOT.jar to /home/hadoop/./opensearch-spark-ppl_2.12-0.3.0-SNAPSHOT.jar
24/03/18 21:45:32 WARN DependencyUtils: Local jar /home/hadoop/skipping does not exist, skipping.
Exception in thread "main" java.lang.IllegalArgumentException: Unsupported number of arguments. Expected 1 or 2 arguments. 5, index on mys3.default.http_logs file:///home/hadoop/.ivy2/jars/org.opensearch_opensearch-spark-sql-application_2.12-0.3.0-SNAPSHOT.jar query_execution_result_mys3
	at org.apache.spark.sql.FlintJob$.main(FlintJob.scala:50)
	at org.apache.spark.sql.FlintJob.main(FlintJob.scala)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:1066)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:192)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:215)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:91)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1158)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1167)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
24/03/18 21:45:32 INFO ShutdownHookManager: Shutdown hook called
24/03/18 21:45:32 INFO ShutdownHookManager: Deleting directory /tmp/spark-b28da86c-b55e-45e6-9aa5-48432a15c19e
```


Example 2:
```
--conf spark.flint.job.query=select * from mys3.default.http_logs limit 10
```
```
Files  s3://flint-dev-seankao/opensearch-spark-jars/opensearch-spark-standalone_2.12-0.3.0-SNAPSHOT.jar from /tmp/spark-89b54a98-e1e7-47dd-849e-a16894ccfa03/opensearch-spark-standalone_2.12-0.3.0-SNAPSHOT.jar to /home/hadoop/./opensearch-spark-standalone_2.12-0.3.0-SNAPSHOT.jar
Files  s3://flint-dev-seankao/opensearch-spark-jars/opensearch-spark-sql-application_2.12-0.3.0-SNAPSHOT.jar from /tmp/spark-89b54a98-e1e7-47dd-849e-a16894ccfa03/opensearch-spark-sql-application_2.12-0.3.0-SNAPSHOT.jar to /home/hadoop/./opensearch-spark-sql-application_2.12-0.3.0-SNAPSHOT.jar
Files  s3://flint-dev-seankao/opensearch-spark-jars/opensearch-spark-ppl_2.12-0.3.0-SNAPSHOT.jar from /tmp/spark-89b54a98-e1e7-47dd-849e-a16894ccfa03/opensearch-spark-ppl_2.12-0.3.0-SNAPSHOT.jar to /home/hadoop/./opensearch-spark-ppl_2.12-0.3.0-SNAPSHOT.jar
24/03/18 21:53:18 WARN DependencyUtils: Local jar /home/hadoop/* does not exist, skipping.
Exception in thread "main" java.lang.IllegalArgumentException: Unsupported number of arguments. Expected 1 or 2 arguments. 8, from mys3.default.http_logs limit 10 --conf spark.executorEnv.JAVA_HOME=/usr/lib/jvm/java-17-amazon-corretto.x86_64/ file:///home/hadoop/.ivy2/jars/org.opensearch_opensearch-spark-sql-application_2.12-0.3.0-SNAPSHOT.jar query_execution_result_mys3
	at org.apache.spark.sql.FlintJob$.main(FlintJob.scala:50)
	at org.apache.spark.sql.FlintJob.main(FlintJob.scala)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:1066)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:192)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:215)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:91)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1158)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1167)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
24/03/18 21:53:18 INFO ShutdownHookManager: Shutdown hook called
24/03/18 21:53:18 INFO ShutdownHookManager: Deleting directory /tmp/spark-89b54a98-e1e7-47dd-849e-a16894ccfa03
```

**Comment by seankao-az** [comment]
```
--conf spark.flint.job.query="refresh skipping index on mys3.default.http_logs"
```

after wrapping the query with quotes, job succeeds

**Pull request merged**